### DISCUSSION TITLE
ZIP issues

### DISCUSSION THREAD

**Discussion by Stoufs** [rejetto/hfs#420] [General]
Is it possible to get someone to zip the content of folders where the files is and not the entire archive? Can't seem to find an option as allowing zip allows you to select every single folder in the home directory and zip them all up

**Comment by rejetto** [comment]
i don't understand what you mean, "the content of folders where the files is" ? what files?
by default the home has permissions to zip, but you can restrict it to only some things

**Comment by Stoufs** [comment]
i disabled home directory being archived but the user can just go in and select all the sub folders in the home directory and zip them. there's no option to zip files in the sub folders only of the home directory itself.

**Comment by rejetto** [comment]
you can do it by setting the permission on the folder content, like this
<img width="326" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/981341be-2db4-46ff-8632-80b42d7171d9">

**Reply by Stoufs** [reply]
Ya but if i set the home directory to say /home and there are sub folders in there, the person goes into the /home directory and selects all the sub folders in there. Thats what i dont want.

**Comment by rejetto** [comment]
you set the permissions for each subfolder, so they won't be able to select all the subfolders

**Reply by Stoufs** [reply]
I have over 1,000 sub folders in there

**Reply by Stoufs** [reply]
If we can get a permission for like > home (No zip; you have that option), > sub folder (no zip) > files (can zip all)

**Comment by rejetto** [comment]
you can do the same that i told you, but for the 1000 subfolders, by using masks.
you can set in the home a mask like
```
"**/*":
  can_archive: true
```

this means that in any subfolder (**) you wan the content (/*) to be archivable.
the "zip" you see in the browser is called "archive" internally.

**Reply by rejetto** [reply]
https://github.com/rejetto/hfs/wiki/Masks-field

**Reply by Stoufs** [reply]
You can still archive sub folders in the home directory unfortunately even with this.

**Reply by Stoufs** [reply]
Ok update! I was able to get it working, but it seems even though the zip option is greyed out in the main directory, the person can still exploit this by selecting the folders they want, hovering over the zip option and copy / paste the link of the zip option in the browser directly and it works fine still so there's a way around this.

**Comment by rejetto** [comment]
ok, i had the time to look into the zip behavior.
The exploit you are talking about, the server is actually incoherent with the interface, searching for anything that you can zip. The good is that you get only things that you can get, that have zip permission, it's not a security problem, but still can be not what the admin wants. The interface came later, and sadly incoherent. Something must be done.

When I made the server behavior like it is now, it looked good, so that "i give no permission for the folder, but for some files, then you can still ask to zip those, and you can do it with a simple button, why not".
While this scenario still look interesting, your scenario is too, so the best would be to fit both.

I guess that I can really prevent the zip of the folder if it has no permission, like you want. People willing to have the other scenario can set permissions in the opposite way: give permission to the folder, then denying for all the content excluding the things to be allowed. This logic can be done starting with this
<img width="284" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/5d30d2cb-4072-489a-9f4a-3ee8169d4818">
Allowing specific things can be done in multple ways, but most of times will be using masks, but if the things are directly accessible in the VFS you can click on every element and set permissions.
You could even do everything just using masks, if you like it.

I think this will be for the best.
I'm working on it.

**Comment by rejetto** [comment]
ok, i made a few adjustments. Will you let me know if looks ok to you?

[hfs-mac-arm.zip](https://github.com/rejetto/hfs/files/13798106/hfs-mac-arm.zip)
[hfs-mac.zip](https://github.com/rejetto/hfs/files/13798107/hfs-mac.zip)
[hfs-linux.zip](https://github.com/rejetto/hfs/files/13798108/hfs-linux.zip)
[hfs-windows.zip](https://github.com/rejetto/hfs/files/13798109/hfs-windows.zip)

**Comment by Stoufs** [comment]
Thats amazing! This works AMAZINGLY now!

**Reply by rejetto** [reply]
Neat!
